apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -224,31 +243,29 @@ defmodule Aecore.Chain.Worker do
     end
   end
 
-  @spec add_block(Block.t()) :: :ok | {:error, String.t()}
-  def add_block(%Block{header: %Header{prev_hash: prev_hash}} = block) do
+  @spec add_block(Block.t(), boolean()) :: :ok | {:error, String.t()}
+  def add_block(
+        %{header: %{prev_hash: prev_hash, prev_key_hash: prev_key_hash, height: height}} = block,
```

Comment:
No type safety here. Add "when block.__struct__ in [KeyBlock, MicroBlock]"
The typespec is wrong.

Reply:
I don't think there's a need for explicit type checking in this case, we do checks in other functions/fragments that are part of this